ibis/util.py

```diff
@@ -522,6 +522,21 @@ def gen_name(namespace: str) -> str:
     return f"ibis_{namespace}_{uid}"
 
 
+def gen_name_from_path(path: str | Path, file_type: str) -> str:
+    """Create a user-friendly unique identifier from a file path.
+
+    Examples
+    --------
+    >>> gen_name_from_path("path/to/myfile.csv", "csv")  # doctest: +ELLIPSIS
+    'ibis_read_csv_myfile_...'
+    """
+    root, _ext = os.path.splitext(path)
+    basename = os.path.basename(root)
+    basename = re.sub(r"[^a-zA-Z0-9_]", "_", basename).strip("_")
+    prefix = f"read_{file_type}_{basename}"
```

Comment:
Since you already have the file extension, do you want to just use that for the `file_type` part of the generated name?

Reply:
Yeah, I think you are right, let's just keep it simple. I actually have another idea on how to make this more user-friendly. Will push changes soon.

Reply:
well, my idea was to use the full path. But MySQL has a limit of 64 chars for table names, so pretty much any name of any reasonabkle length is too long. So I kept it to just use the basename as it currently does.